clippy_lints/src/loops/manual_memcpy.rs

```diff
@@ -186,11 +189,13 @@ fn build_manual_memcpy_suggestion<'tcx>(
         "clone_from_slice"
     };
 
-    format!(
-        "{dst}.{method_str}(&{src_base_str}[{}..{}]);",
-        src_offset.maybe_par(),
-        src_limit.maybe_par()
-    )
+    let src = if is_array_length_equal_to_range(cx, start, end, src.base) {
+        src_base_str
+    } else {
+        format!("{src_base_str}[{}..{}]", src_offset.maybe_par(), src_limit.maybe_par()).into()
```

Comment:
This `.into()` can be converted to a reference (If it's converting that `format!` to `&str`, which I think that it's doing).

```suggestion
        &format!("{src_base_str}[{}..{}]", src_offset.maybe_par(), src_limit.maybe_par())
```

Reply:
I appreciate your review, but converting `format!("{src_base_str}[{}..{}]", src_offset.maybe_par(), src_limit.maybe_par()).into()` to `&format!("{src_base_str}[{}..{}]", src_offset.maybe_par(), src_limit.maybe_par())` does not work because it requires `Cow<'_, str>`, so I didn't apply the suggestion.
If you have other ideas, I'm open to making the necessary changes.